tests/lib/config/config.js

```diff
@@ -504,14 +504,62 @@ describe("Config", () => {
 			});
 
 			const json = config.toJSON();
+			assert.deepStrictEqual(json, {
+				plugins: ["test:testPlugin@1.0.0"],
+				language: "test/lang",
+				processor: "testProcessor@1.0.0",
+				languageOptions: {
+					ecmaVersion: 2022,
+					sourceType: "module",
+					parser: "testParser",
+				},
+			});
+		});
 
-			assert.deepStrictEqual(json.plugins, ["test:testPlugin@1.0.0"]);
-			assert.strictEqual(json.language, "test/lang");
-			assert.strictEqual(json.processor, "testProcessor@1.0.0");
-			assert.deepStrictEqual(json.languageOptions, {
-				ecmaVersion: 2022,
-				sourceType: "module",
-				parser: "testParser",
+		it("should serialize when a language option has a toJSON() method and a function", () => {
+			const mockLanguage = {
+				validateLanguageOptions() {},
+				normalizeLanguageOptions(options) {
+					options.syntax.toJSON = () => "syntax";
+					return options;
+				},
+				meta: {
+					name: "testLang",
+					version: "1.0.0",
+				},
+			};
+
+			const mockPlugin = {
+				meta: {
+					name: "testPlugin",
+					version: "1.0.0",
+				},
+				languages: {
+					lang: mockLanguage,
+				},
+			};
+
+			const config = new Config({
+				language: "test/lang",
+				plugins: {
+					test: mockPlugin,
+				},
+				languageOptions: {
+					syntax: {
+						atrule: {
+							name() {},
+						},
+					},
+				},
+			});
+			const json = config.toJSON();
+			assert.deepStrictEqual(json, {
+				plugins: ["test:testPlugin@1.0.0"],
+				processor: void 0,
+				language: "test/lang",
+				languageOptions: {
+					syntax: "syntax",
+				},
 			});
```

Comment:
I think we can add a few more tests:
- when `toJSON()` is not a function.
- Multiple Nested toJSON() Calls
    ```js
    syntax: {
      block: {
        selector: {
          toJSON: () => "selector",
        },
        toJSON: () => "block",
      },
    }
    ```
- `toJSON()` Returns a Non-Serializable Value

    ```js
    toJSON: () => () => "I am a function"  // or a circular object
    ````

Reply:
Are you saying we should always check the value returned from `toJSON()` to ensure it's serializable? And if not throw an error?

Reply:
Yes, just making sure we are providing a friendly error message.

Reply:
Maybe we can say that, by adding `.toJSON` methods, the Language takes all responsibility for the proper serialization of returned values.

Reply:
I updated to just check if the returned value is a function, which gives it the same check as any other key in `languageOptions`. I think that should suffice.